### PR TITLE
Add status flag to Gallery items

### PR DESCRIPTION
**Pull request by flamerohr**
So there's a mark for if an item is modified or in draft-only. (top left dot)

Also added a label next to the dimensions to show some text if it is modified or draft-only.

screen shot 2016-10-04 at 8 58 07 pm

Requires these before merging:
 - [ ] https://github.com/silverstripe/silverstripe-framework/pull/6136
 - [ ] https://github.com/silverstripe/silverstripe-cms/pull/1637

### CODE REVIEW

**Comment by tractorcow** [comment]
I've added some cleanup to the versioned status flags checks, and created a new PR for CMS changes.

File: code/Controller/AssetAdmin.php [hunk 1]
```diff
@@ -571,13 +570,13 @@ public function getFileEditForm($id)
         if (!$file->canView()) {
             $this->httpError(403, _t(
                 'AssetAdmin.ErrorItemPermissionDenied',
-                'It seems you don\'t have the necessary permissions to add {ObjectTitle} to a campaign',
+                'You don\'t have the necessary permissions to modify {ObjectTitle}',
```
Comment (tractorcow) [inline code/Controller/AssetAdmin.php]: Will do.

File: code/Controller/AssetAdmin.php [hunk 2]
```diff
@@ -875,9 +874,9 @@ public function getAddToCampaignForm($id)
         if (!$record->canView()) {
             $this->httpError(403, _t(
                 'AssetAdmin.ErrorItemPermissionDenied',
-                'It seems you don\'t have the necessary permissions to add {ObjectTitle} to a campaign',
+                'You don\'t have the necessary permissions to modify {ObjectTitle}',
```
Comment (tractorcow) [inline code/Controller/AssetAdmin.php]: Done.

**Comment by tractorcow** [comment]
@ingo ready for re-review.

**Comment by chillu** [comment]
Tests are still failing because of linting errors

**Pull request merged**